waku/v2/node/waku_node.nim

```diff
@@ -371,47 +373,32 @@ proc startRelay*(node: WakuNode) {.async.} =
   info "relay started successfully"
 
 proc mountRelay*(node: WakuNode,
-                 topics: seq[string] = newSeq[string](),
+                 topics: seq[string] = @[],
                  triggerSelf = true,
-                 peerExchangeHandler = none(RoutingRecordsHandler))
-  # TODO: Better error handling: CatchableError is raised by `waitFor`
-  {.async, gcsafe, raises: [Defect, InitializationError, LPError, CatchableError].} =
-
-  proc msgIdProvider(m: messages.Message): Result[MessageID, ValidationResult] =
-    let mh = MultiHash.digest("sha2-256", m.data)
-    if mh.isOk():
-      return ok(mh[].data.buffer)
-    else:
-      return ok(($m.data.hash).toBytes())
-
-  let wakuRelay = WakuRelay.init(
-    switch = node.switch,
-    msgIdProvider = msgIdProvider,
-    triggerSelf = triggerSelf,
-    sign = false,
-    verifySignature = false,
-    maxMessageSize = MaxWakuMessageSize
+                 peerExchangeHandler = none(RoutingRecordsHandler)) {.async, gcsafe.} =
```

Comment:
> Since mountRelay does not raise any exception now. Should we change also [this](https://github.com/status-im/nwaku/blob/master/apps/wakunode2/wakunode2.nim?rgh-link-date=2022-11-18T08%3A07%3A34Z#L346-L350). There is a try but no exceptions are raised.

Yes

> Slightly related, should we have a return value? so if initRes.isErr() return err() and the caller can be aware that it failed? Perhaps just Result[void, string]?

Yes, something like that. And all those changes should go together under the same PR. This is out of the scope of this PR.

------

FYI There are some changes planned that will rework that relationship `waku_node <-> wakunode2`:

```markdown
# Waku Node

- Add Waku node builder (mimic libp2p switch builders) to solve the “protocol mounting” and “switch start” ordering issues.
- TBD
```

What you are saying is included in the TBD. Sorry, but I cannot specify everything with details without the proper analysis 😅

Reply:
if its tracked, im happy :) resolving